astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -23,212 +168,638 @@ class NDArithmeticMixin(object):
+        if len(self.meta) == 0 and len(operand.meta) > 0:
+            # The only case in which we would like the meta from the second
+            # operand is if the first operand has no meta and the second one
+            # contains at least one meta element.
+            # There is no use-case where merging the meta of both operands
+            # makes sense. In case one might really set the results meta
+            # information to match the one of the second operand is to put a
+            # reference or copy inside the parameter that sets the keywords
+            # in here.
+            result_meta = deepcopy(operand.meta)
+        else:
+            result_meta = deepcopy(self.meta)
+
+        # Now get to work, first we check if there are any keywords which
+        # should be arithmetically operated upon.
+        if meta_kwds_operate is not None and len(meta_kwds_operate) > 0:
+
+            # Get the operator function for these operation (use numpy ufuncs)
+            if operation == 'addition':
+                operator = np.add
+            elif operation == 'subtraction':
+                operator = np.subtract
+            elif operation == 'multiplication':
+                operator = np.multiply
+            elif operation == 'division':
+                operator = np.true_divide
+            else:
+                raise ValueError('Unsupported operation')
+
+            # There are two possibilities, either operate keyword of operand1
+            # with the same keyword of operand2. Like adding exposuretimes in
+            # addition. So if both operands have meta information assume
+            # this kind is wanted. The other possibility is to operate
+            # the keyword of the only meta (since if both have meta we try
+            # the other way) with the number of the other operand. For example
+            # exposure time is multiplied with 5 if the data is multiplied by
+            # 5.
+            if len(result_meta) == 0:
+                # No meta present. Nothing to do but issueing a warning that
+                # it was not possible.
+                log.info("No meta is present so no operation on them is "
+                         "possible.")
+            elif len(self.meta) > 0 and len(operand.meta) > 0:
+                # Both have meta information so we assume that we operate
+                # keyword on keyword.
+                for i in meta_kwds_operate:
+                    result_meta[i] = operator(self.meta[i], operand.meta[i])
+            elif len(self.meta) > 0:
+                # We have one meta, now we need to operate the keywords with
+                # the second operands data. But that's only well defined if
+                # that is a scalar... (ok, at least if we don't want to allow
+                # the meta to contain numpy arrays :-) )
+                if operand.data.size != 1:
+                    log.info("The second operand is not a scalar. Cannot "
```

Comment:
I think this should raise an exception instead of logging an information message.